Körtagság listázás api

tmichel opened the pull request:
Alapvetően a körtagságok listázásához szükséges dolgok vannak itt, némi becsúszott refaktorálással.
- `@EJB`-ket `@Inject`-re cseréltem
- körkörös injektálást feloldottam (csak bent maradt változók voltak)
- a membershipeket betöltő query-t kiigazítottam, hogy ne `n+1` query-ben töltse le az adatbázisból a tagságokat és a hozzá tartozó adatokat.
- UserManagerBean teszt barátabbá tétele (setter injectionnel)

Hosszabb távon majd az `EjbConstructorArgument` osztályt fel kell számolni. Ez már anno is hacknek tűnt.

tmichel committed:
membership namedquery cleanup

tmichel committed:
replaced EJB annotation with cdi's Inject (#97)

tmichel committed:
listing posts in membership, and tests

tmichel committed:
fetching memberships with groups and posts

mini-refactor of UserManagerBean to use setter injection

tmichel committed:
making AbstractEntityView#clazz final

tmichel committed:
memberships api endpoint

listing
* membership start, end
* group name
* posts as a string list (including öregtag and tag)

tmichel committed:
using maven resource filtering for persistence.xml configuration

hibernate.show_sql is true for development but false for production

vbalazs committed:
fix typo in install.md

vbalazs commented on sch-pek-domain/src/main/java/hu/sch/domain/Membership.java:
```diff
@@ -211,26 +207,29 @@ public int hashCode() {
+    public List<String> getAllPosts() {
+        List<String> postList = new ArrayList<>();
+        for (Post post : getPosts()) {
+            postList.add(post.getPostType().getPostName());
```
ezt szerintem lehetne java8 szintaxissal szebben is.
meg kene nezni a forEach-et vagy esetleg a map-et.

tmichel replied:
igen. de egyelőre még 7es jdk-n futunk. akkor álljunk át 8-asra? :smile:

vbalazs replied:
mukodnie kell siman :) :rocket:

tmichel replied:
igen, de ha kitelepítjük, akkor a többi javas cuccunk (openam?) elmegy vele?

vbalazs replied:
az irc-n megbeszeltek szerint ezzel nem kell foglalkozni

vbalazs commented on sch-pek-domain/src/main/java/hu/sch/domain/user/User.java:
```diff
@@ -393,14 +393,14 @@ public void sortMemberships() {
         if (this.getMemberships() != null) {
             Collections.sort(this.getMemberships(),
                     new Comparator<Membership>() {
```
Tudom, hogy ez itt volt eddig is, de szerintem ezt a comparatort ki kene emelni egy kulon classba. Nem a User-be valo. Vagy a Membership-nek kellene a comparable-t implementalnia? Utobbi csak otlet, azert nem egyertelmu.

tmichel replied:
Igen. Ezzel jó lenne valamit kezdeni. Bár igazából az egész `sortMemberships`-et ki lehetne emelni.

vbalazs replied:
jogos. az plane nem valo a Userbe

tmichel committed:
introduces MembershipSorter

sortMembership() is removed from User and Group entities.
Memberships are sorted in the api endpoint.

tmichel committed:
upgraded to jdk8

tmichel committed:
using the new streaming api for mapping collections

tmichel committed:
annotations in Membership and User

User#sviePrimaryMembership is set to lazy fetch
